llrt_modules/src/modules/zlib.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use std::io::Read;
+
+use brotlic::{CompressorReader as BrotliEncoder, DecompressorReader as BrotliDecoder};
+use flate2::{
+    read::{DeflateDecoder, DeflateEncoder, GzDecoder, GzEncoder, ZlibDecoder, ZlibEncoder},
+    Compression,
+};
+use llrt_utils::{ctx::CtxExtension, module::export_default, object::ObjectExt, result::ResultExt};
+use rquickjs::function::Func;
+use rquickjs::{
+    module::{Declarations, Exports, ModuleDef},
+    prelude::{Opt, Rest},
+    Ctx, Error, Exception, FromJs, Function, IntoJs, Null, Object, Result, Value,
+};
+use zstd::{
+    stream::read::{Decoder as ZstdDecoder, Encoder as ZstdEncoder},
+    zstd_safe::CompressionLevel,
+};
+
+use crate::{utils::array_buffer::ArrayBufferView, ModuleInfo};
+
+use super::buffer::Buffer;
+
+macro_rules! define_sync_function {
+    ($fn_name:ident, $converter:expr, $command:expr) => {
+        pub fn $fn_name<'js>(
+            ctx: Ctx<'js>,
+            value: Value<'js>,
+            options: Opt<Object<'js>>,
+        ) -> Result<Value<'js>> {
+            $converter(ctx.clone(), value, options, $command)
+        }
+    };
+}
+
+macro_rules! define_cb_function {
```

Comment:
Thats like a very cool macro, do you think we could generalize it and put it in the util? I know fs is missing all the functions with callbacks and that could help

Reply:
If it is to be generalized, I would need to make additional considerations regarding the number of parameters other than callback functions.

I have done my best to create a macro for my current use case, but after this PR is merged, I would welcome this macro to be further generalized to work elsewhere. :)